### PR TITLE
neetoUI v6

### LINKED ISSUE #1854: Enhance customizability of neetoUI
Currently, neetoUI doesn't support many customization options. 

### Proposed solution

Enhancing the customizability of neetoUI by implementing a set of customization features similar to popular design systems like [Bootstrap](https://getbootstrap.com/docs/5.3/customize/overview/) and [Ant design](https://ant.design/docs/react/customize-theme). These features would empower users to adapt the library to their unique design without compromising on the core advantages of using neetoUI.

### Customization features to consider

- **Color theming** : Allow users to define custom color palettes or choose from predefined themes.
- **Component styling** : Enable developers to easily override default styles for specific components.
- **Responsive Typography** : Provide support for responsive Typography - [RFS](https://getbootstrap.com/docs/5.3/getting-started/rfs/#what-is-rfs).
- **Responsive paddings/margins** : Provide support for responsive paddings and margins - [RFS](https://getbootstrap.com/docs/5.3/getting-started/rfs/#what-is-rfs).
- **Global CSS variables** : Expose global CSS variables for further fine-tuning.

@praveen-murali-ind _A

### PR DESCRIPTION
**Pull request by goutham-subramanyam**
- Fixes #1854 

**Description**
- Changed: Enhanced the customizability of all the components by allowing users
  to easily override the default styles.
- Added: Added CSS customization storybook example for all the components.
- Changed: Updated _Typography_ component to be responsive using
  [RFS](https://github.com/twbs/rfs/tree/v9.0.3).
- Changed: Updated components using Ant Design to CSS-in-JS approach with
  improved dark mode support.
- Removed: Removed Ant Design vendor styles in favor of CSS-in-JS approach.
- Removed: Removed unused components and styles.
- Fixed: Fixed _TimePicker_ component dark mode issue.

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] ~I have updated the types definition of modified exports.~
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] ~I have added tests that prove my fix is effective or that my feature works~
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

**Reviewers**

### CODE REVIEW

**Comment by praveen-murali-ind** [comment]
@goutham-subramanyam _a @ajmaln Let’s resolve the conflicts and merge the v6 today.

**Comment by goutham-subramanyam** [comment]
@ajmaln _a Please review.

File: stories/Customize/CSSVariables.stories.mdx [hunk 1]
```diff
@@ -24,111 +24,7 @@ Here are the variables we include that are accessible from any location where ne
...
-  --neeto-ui-surface-bg: 255, 255, 255;
-  --neeto-ui-pastel-color-1: 229, 231, 235;
-  --neeto-ui-pastel-color-2: 254, 202, 202;
-  --neeto-ui-pastel-color-3: 253, 230, 138;
-  --neeto-ui-pastel-color-4: 167, 243, 208;
-  --neeto-ui-pastel-color-5: 191, 219, 254;
-  --neeto-ui-pastel-color-6: 199, 210, 254;
-  --neeto-ui-pastel-color-7: 221, 214, 254;
-  --neeto-ui-pastel-color-8: 251, 207, 232;
+
```
Comment (ajmaln) [inline stories/Customize/CSSVariables.stories.mdx:27]: Shouldn't include the variables here?

**Review by ajmaln — COMMENTED** [review]
@goutham-subramanyam _a Few suggestions

File: src/components/ColorPicker/Picker.jsx
[the remark this answers is not in the document]
Comment (goutham-subramanyam) [reply src/components/ColorPicker/Picker.jsx:1]: Yes.

[the remark this answers is not in the document]
Comment (goutham-subramanyam) [reply src/components/ColorPicker/Picker.jsx:1]: Ref: https://github.com/bigbinary/neeto-ui/pull/1392

File: stories/Customize/CSSVariables.stories.mdx (hunk 1 above)
Comment (goutham-subramanyam) [reply stories/Customize/CSSVariables.stories.mdx:27]: The component-specific variables are listed at the bottom of the respective components with an override example.

Example:

Screenshot 2023-12-01 at 11 02 40 PM

**Comment by goutham-subramanyam** [comment]
@ajmaln _a I have made the suggested changes. Please check. We can merge the changes by Monday morning as discussed with @praveen-murali-ind.

**Comment by goutham-subramanyam** [comment]
@ajmaln _a I have fixed the ESLint errors. Please check.

**Review by ajmaln — APPROVED** [review]
LGTM

**Pull request merged**